test/es6/default-splitscope.js

```diff
@@ -6,260 +6,622 @@
+           })(); 
+
+           (function (a = 10, b = function () { assert.areEqual(a, 10, "Function defined in the param scope captures the formals from the param scope when executed from the param scope"); }, c = b()) { 
+           })(); 
+
+           function f4(a = 10, b = function () { return a; }) { 
+               a = 20; 
+               return b; 
+           } 
+           assert.areEqual(f4()(), 10, "Even if the formals are not redeclared in the function body the symbol in the param scope and body scope are different"); 
+
+           function f5(a = 10, b = function () { return function () { return a; }; }) { 
+               var a = 20; 
+               return b; 
+           } 
+           assert.areEqual(f5()()(), 10, "Parameter scope works fine with nested functions"); 
+
+           var a1 = 10; 
+           function f6(b = function () { return a1; }) { 
+               assert.areEqual(a1, undefined, "Inside the function body the assignment hasn't happened yet"); 
+               var a1 = 20; 
+               assert.areEqual(a1, 20, "Assignment to the symbol inside the function changes the value"); 
+               return b; 
+           } 
+           assert.areEqual(f6()(), 10, "Function in the param scope correctly binds to the outer variable"); 
+            
+           function f7(a = 10, b = { iFnc () { return a; } }) { 
+               a = 20; 
+               return b; 
+           } 
+           assert.areEqual(f7().iFnc(), 10, "Function definition inside the object literal should capture the formal from the param scope"); 
+       } 
+   }, 
+   { 
+       name: "Split parameter scope in function expressions with name", 
+       body: function () { 
+           function f1(a = 10, b = function c() { return a; }) { 
+               assert.areEqual(a, 10, "Initial value of parameter in the body scope of the method should be the same as the one in param scope"); 
+               var a = 20; 
+               assert.areEqual(a, 20, "New assignment in the body scope of the method updates the variable's value in body scope"); 
+               return b; 
+           } 
+           assert.areEqual(f1()(), 10, "Function expression defined in the param scope captures the formals from the param scope not body scope"); 
             
-            assert.throws(function () { eval("function foo(a = eval('b'), b) {}; foo();"); }, ReferenceError, "Future default references using eval are not allowed", "Use before declaration");
+           function f1(a = 10, b = function c(recurse = true) { return recurse ? c(false) : a; }) { 
+               return b; 
+           } 
+           assert.areEqual(f1()(), 10, "Function expression defined in the param scope captures the formals from the param scope not body scope"); 
+       } 
+   }, 
+   { 
+       name: "Split parameter scope in member functions", 
```

Comment:
Is there a difference between function declarations and object literal concise function members that affects codes paths related to split scope?
This test case feels redundant to me but maybe I'm missing something.